### ISSUE TITLE
files can not remove from db

### ISSUE THREAD

**Issue by sugen** [RocketChat/Rocket.Chat#7455]
Rocket.Chat Version: 0.57.1
Running Instances: centos

when i upload the file to the system by rocket chat, but i want to remove it , when i remove it , the screen was removed but the mongodb still has it , the file have not remove from the db, so how to remove it both?

**Comment by pkgodara** [comment]
WIP #7383 .

**Comment by geekgonecrazy** [comment]
@sugen Do you have save message edit history on?

**Comment by geekgonecrazy** [comment]
@MartinSchoeler / @pkgodara this seems like a case where the file was deleted... but its not being deleted in mongo.

I had a case like this in [#ubuntu-snap](https://demo.rocket.chat/channel/ubuntu-snap) the issue ended up being that they had save message history on.  

So it just modified the file in mongodb and added the `_hidden: true` property to it.  The same thing I believe it does on the message object.

**Comment by geekgonecrazy** [comment]
Going to go ahead and close this.  Please try to disable message edit history if it doesn't fix please report back and we can investigate from there

**Comment by Julianoe** [comment]
Hi i have this exact same issue.
One of my users uploaded 2 large files (130mo) as a test of our chat. My complete mongodump weight approximatly 520mo (not compressed). So i'm thinking: "okay those files represent a large part of our database, i sould delete them now".

I did that via the attachment interface but i did another mongodump it was the exact same size. I've been to the the Message admin panel and disabled Message_KeepHistory (**in a UI perspective that red arrow by the way is really confusing**). It did not change anything. I also tried the API_embed_clear_cache_now but i does not seem to be related.

Can we investigate this?

**Comment by geekgonecrazy** [comment]
that makes sense for that file to big biggest as thats what contains actual files uploaded.

Did you disable the setting before or after you deleted the message?

**Comment by Julianoe** [comment]
I'm not sure to understand the question but my actual question is simple: how can we delete a file for real? If it's deleted from everywhere, why is still the database the same size?

**Comment by geekgonecrazy** [comment]
Let me rephrase...

Did you delete the file before or after you turned off keep message history?

If you turned off after... The file was not actually deleted from the db.

If you turned off before then something else might be up.

To prune... Take a look at the rocketchat_uploads collection.  

Now another thing... With mongo just because you delete something doesn't mean you gain the space back.  It allocates space to increase write performance.  If you export, drop db, re-import you can get real size.
